### PR TITLE
#164069189 Social Share

### PR DESCRIPTION
**Pull request by wasibani-roy**
**What does this PR do?**
This PR enables the user of our system to share articles they love via social media or email

**Description of Task to be completed?**

- User should be able to share an article via social media
- User should be able to share an article via email

**How should this be manually tested?**

- Clone the repo and cd into ah-aquaman-frontend
- checkout the branch `ft-social-share-164069189`
- Run the commands below to install packages and run the server
         `npm install`
         `npm start`
- Navigate to the article you wish to share and click on any of the social share buttons

**What are the relevant pivotal tracker stories?**

- [#164069189](https://www.pivotaltracker.com/story/show/164069189)

**Screenshots**
Screenshot 2019-04-23 at 09 12 14

### CODE REVIEW

**Review by CryceTruly — CHANGES_REQUESTED** [review]
Hey @wasibani-roy, great work, I love the implementation. However I have a problem sharing to email, facebook and twitter, It looks like the app is unable to pick my current domain, What could be the problem.
Screenshot 2019-04-23 at 12 16 26
Screenshot 2019-04-23 at 12 17 44
Screenshot 2019-04-23 at 12 19 23

**Review by CryceTruly — APPROVED** [review]

**Review by Peace-Apple — APPROVED** [review]
Great work @wasibani-roy

**Review by joelethan — APPROVED** [review]

File: src/routes/index.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const Routes = () => (
       <Route path="/signup" exact component={signupView} />
       <ProtectedRoute path="/profile" exact component={Profile} />
       <ProtectedRoute path="/editprofile" exact component={EditProfile} />
-      <ProtectedRoute path="/article/:slug" exact component={ArticleDetailView} />
+      <Route path="/article/:slug" exact component={ArticleDetailView} />
```
Comment (AmosWels) [inline src/routes/index.js:26]: why have you removed the protected route component

File: src/views/articleDetailView/index.js [hunk 2]
```diff
@@ -8,6 +8,17 @@ import Sidebar from "../../components/article/sidebar";
...
+  const articleLink = `${process.env.FRONTEND_BASE_URL}/article/${slug}`;
+  const shareBody = `I found this article to be worth reading, ${article.title} ${articleLink}`;
+  const mailshare = `mailto:?Subject=${encodeURI(article.title)}&body=${shareBody}`;
+  const facebookShare = `https://www.facebook.com/sharer/sharer.php?u=${encodeURI(articleLink)}`;
+  const twitterShare = `https://twitter.com/home?status=${encodeURI(shareBody)}`;
+
+  return {
+    mailshare, facebookShare, twitterShare,
+  };
+}
```
Comment (AmosWels) [inline src/views/articleDetailView/index.js:21]: leave an extra line here

**Review by AmosWels — CHANGES_REQUESTED** [review]
Goodwork @wasibani-roy, implement the feedback on your PR

File: src/routes/index.js (hunk 1 above)
Comment (wasibani-roy) [reply src/routes/index.js:26]: We decided as a team that one should be able to read an article/view even when not logged in but can not make any changes to it until they log in

File: src/views/articleDetailView/index.js (hunk 2 above)
Comment (wasibani-roy) [reply src/views/articleDetailView/index.js:21]: Thank you I have added the line

**Review by AmosWels — APPROVED** [review]

**Pull request merged**